source/data-formats/extended-json.txt

```diff
@@ -0,0 +1,203 @@
+store data.
+
+Extended JSON Formats
+---------------------
+
+MongoDB Extended JSON features different string formats to represent BSON data.
+Each of the different formats conform to the JSON RFC
+and meet specific use cases. The **Extended** format, also known as the
+**Canonical** format, features specific representations for every BSON type
+for bidirectional conversion without loss of information. The **Relaxed Mode**
+format is more concise and closer to ordinary JSON, but does not represent
+all the type information such as the specific byte size of number fields.
+
+See the following table to see a description of each format:
+
+.. list-table::
+   :header-rows: 1
+   :stub-columns: 1
+   :widths: 10 40
+
+   * - Name
+     - Description
+
+   * - **Extended**
+     - | Also known as the *Canonical* format, this JSON representation avoids loss of
+         BSON type information.
+       | This format prioritizes type preservation at the loss of human-readability and
+         interoperability with older formats.
+
+   * - **Relaxed Mode**
+     - | JSON representation that describes BSON documents with some type information loss.
+       | This format prioritizes human-readability and interoperability at the loss of
+         certain type information.
+
+To learn more about JSON, BSON, and Extended JSON, see
+`our article about JSON and BSON <https://www.mongodb.com/resources/basics/json-and-bson>`__
+and :manual:`Extended JSON </reference/mongodb-extended-json/>` in the {+mdb-server+} manual.
+
+.. _extended_json_example_section:
+
+Extended JSON Examples
+~~~~~~~~~~~~~~~~~~~~~~
+
+The following examples show a document containing an ObjectId, date, and long
+number field represented in the Extended JSON format. Click the tab that
+corresponds to the format of the example you want to see:
+
+.. tabs::
+
+   .. tab:: Extended
+      :tabid: extended-format
+
+      .. code-block:: json
+
+         {
+           "_id": { "$oid": "573a1391f29313caabcd9637" },
+           "createdAt": { "$date": { "$numberLong": "1601499609" }},
+           "numViews": { "$numberLong": "36520312" }
+         }
+
+   .. tab:: Relaxed Mode
+      :tabid: relaxed-mode-format
+
+      .. code-block:: json
+
+         {
+           "_id": { "$oid": "573a1391f29313caabcd9637" },
+           "createdAt": { "$date": "2020-09-30T18:22:51.648Z" },
+           "numViews": 36520312
+         }
+
+Write Extended JSON
+-------------------
+
+You can write an Extended JSON string from a BSON document object by using the
+``EJSON.stringify()`` method.
+
+The following example outputs an Extended JSON string in the Relaxed format:
+
+.. io-code-block::
+
+   .. input::
+      :language: js
+
+      const { Binary, Code, EJSON } = require('bson');
+
+      doc = [
+         {"foo": [1, 2]},
+         {"bar": {"hello": "world"}},
+         {"code": new Code("function x() { return 1; }", {})},
+         {"bin": Binary.createFromBase64("AQIDBA==", 0)}
+      ]
```

Comment:
```js
      const doc = [
        { foo: [1, 2] },
        { bar: { hello: "world" } },
        { code: new Code("function x() { return 1; }", {}) },
        { bin: Binary.createFromBase64("AQIDBA==", 0) },
      ];
```

Just making this look more JavaScript-y, there's nothing wrong with quoted key syntax but most programmers won't take the time to do it unless there's a key with a symbol in it that can't be written into code literally (like keys with dots or dashes).

Reply:
```suggestion
      const doc = {
        foo: [1, 2],
        bar: { hello: "world" },
        code: new Code("function x() { return 1; }", {}),
        bin: Binary.createFromBase64("AQIDBA==", 0),
    };
```
_Edit:_

Node's EJSON stringifier supports arrays as a top-level object to have parity with the native `JSON.stringify`. In BSON however, and for other drivers, there is no such thing as a top-level array, we should stick to document examples as the top-level argument.